core/store/migrate/plugins/relayer/evm/resolver.go

```diff
@@ -0,0 +1,90 @@
+package evm
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils/big"
+)
+
+// 4 digit version prefix to match the goose versioning
+//
+//go:embed [0-9][0-9][0-9][0-9]_*.tmpl.sql
+var embeddedTmplFS embed.FS
+
+var MigrationRootDir = "."
+
+type Cfg struct {
+	Schema  string
+	ChainID *big.Big
+}
+
+var migrationSuffix = ".tmpl.sql"
+
+func resolve(out io.Writer, in string, val Cfg) error {
+	id := fmt.Sprintf("init_%s_%s", val.Schema, val.ChainID)
+	tmpl, err := template.New(id).Parse(in)
+	if err != nil {
+		return fmt.Errorf("failed to parse template %s: %w", in, err)
+	}
+	err = tmpl.Execute(out, val)
+	if err != nil {
+		return fmt.Errorf("failed to execute template %s: %w", in, err)
+	}
+	return nil
+}
+
+func generateMigrations(fsys fs.FS, rootDir string, tmpDir string, val Cfg) ([]string, error) {
+	err := os.MkdirAll(tmpDir, os.ModePerm)
+	if err != nil {
+		return nil, err
+	}
+	b := make([]byte, 1024*1024)
```

Comment:
I'm not sure how to do this cleanly off the top of my head, but I assume once we get past the initial draft stage, we'd want to make this able to handle arbitrarily large migration files. If not, we should at least make it a constant or config option so we can increase it later if needed.